### DISCUSSION TITLE
Regarding publishing the developed application

### DISCUSSION THREAD

**Discussion by manuks481** [tethysplatform/tethys#681] [Q&A]
Dear developers,

I have been following your tutorial on developing a Tethys App using Google Earth Engine. I have successfully completed the production installation. I have few doubts regarding publishing the app through a public domain name.

1. While purchasing a public domain name, do I have to look into any specific requirements like support for Django? Will it work in a single shared hosting available @ hostinger.com? Or, Can you suggest any web hosting service suitable for Tethys apps?
2. Can you explain the procedure for publishing the app to a public domain name? Is it only setting the ALLOWED_HOSTS in portal_config.yml to the purchased public domain name?

I don't have any experience in creating websites. That's why I'm posting such silly queries. 😄 
I shall be grateful if you could kindly clear my doubts.

Regards

Manu

**Comment by swainn** [comment]
Hi @manuks481:

Thanks for reaching out. Deploying Tethys to production remains one of the biggest hurdles for our users. That said here are some tips:

> 1. While purchasing a public domain name, do I have to look into any specific requirements like support for Django? Will it work in a single shared hosting available @ hostinger.com? Or, Can you suggest any web hosting service suitable for Tethys apps?

I'm not too familiar with hostinger.com, but a quick look at the Single Shared Hosting that you suggested doesn't seem like it will work with Tethys. You'll need SSH access so you can log on to the server and install Tethys. It looks like their Premium Shared Hosting provides that, but I wouldn't know for sure. You'll find a lot of hosting services that are focused on Wordpress sites. Those usually don't give you the access to the server that you need to install Tethys.

What you want to look for is a service that gives you full access to a Virtual Machine. One provider that I like is called Digital Ocean. It's fairly easy to use and their prices are hard to beat. Their basic "Droplets" (VMs) start at $5 per month (see: https://www.digitalocean.com/pricing). For a minimal Tethys server, I'd recommend getting something with at least 2 CPUs, 4 GB of RAM, and 50GB of disk space, which would be the $20 per month option. What you will need will depend on what your app is doing though. You may need more RAM or CPUs if your app is doing a lot of heavy computing or more storage if it is generating/using large datasets. There are many providers like Digital Ocean, though.

Any of the major cloud service providers will have similar options (Amazon Web Services, Microsoft Azure, and Google Cloud). It's quite a bit more difficult to navigate these systems because they have so many products and they aren't named very intuitively (e.g. EC2 = Virtual Machine in Amazon Web Services).

> 2\. Can you explain the procedure for publishing the app to a public domain name? Is it only setting the ALLOWED_HOSTS in portal_config.yml to the purchased public domain name?

Detailed instructions for setting up Tethys in production can be found here: http://docs.tethysplatform.org/en/stable/installation/production.html

We've also created a free course at https://tethys.edunext.io/ that walks through performing an installation on a CentOS virtual machine so you can try it out before getting a server: https://tethys.edunext.io/courses/course-v1:tethys+PRD-100+2020/about

Hope this helps,

Nathan

**Reply by manuks481** [reply]
Hi @swainn,

Thanks much for the detailed reply. This would be really helpful for me.
